generic-sensor/generic-sensor-tests.js

```diff
@@ -141,11 +175,15 @@ function runGenericSensorTests(sensorType) {
     }
   }, "no exception is thrown when calling stop() on already stopped sensor");
 
-  async_test(t => {
-    window.onmessage = t.step_func(e => {
-      assert_equals(e.data, "SecurityError");
-      t.done();
-    });
+  promise_test(t => {
+    return load_iframe(make_script(sensorType))
+      .then(iframe => {
+        iframe.contentWindow.postMessage({}, '*');
```

Comment:
Could there be a race condition? E.g., reply is received from iframe's postMessage before parent is subscribed. Also, this test fails with timeout error.